skyvern/forge/sdk/log_artifacts.py

```diff
@@ -0,0 +1,168 @@
+import json
+import structlog
+
+from skyvern.forge import app
+from skyvern.forge.sdk.core import skyvern_context
+from skyvern.forge.skyvern_json_encoder import SkyvernJSONLogEncoder
+from skyvern.forge.skyvern_log_encoder import SkyvernLogEncoder
+from skyvern.forge.sdk.artifact.models import ArtifactType, LogEntityType
+
+LOG = structlog.get_logger()
+
+def primary_key_from_log_entity_type(log_entity_type: LogEntityType) -> str:
+    if log_entity_type == LogEntityType.STEP:
+        return "step_id"
+    elif log_entity_type == LogEntityType.TASK:
+        return "task_id"
+    elif log_entity_type == LogEntityType.WORKFLOW_RUN:
+        return "workflow_run_id"
+    elif log_entity_type == LogEntityType.WORKFLOW_RUN_BLOCK:
+        return "workflow_run_block_id"
+    else:
+        raise ValueError(f"Invalid log entity type: {log_entity_type}")
+
+async def save_step_logs(step_id: str) -> None:
+    log = skyvern_context.current().log
```

Comment:
```suggestion
    context = skyvern_context.current()
    if not context:
        return
    log = context.log
```

Reply:
skyvern_context.current() can return None.

we still use python shell/script sometimes so it's better to check null here

Reply:
if we want to ensure there's a context, can also use `skyvern_context.ensure_context()`